dev/AppNotifications/AppNotificationDevicesData.cpp

```diff
@@ -0,0 +1,46 @@
+#include "pch.h"
+#include "AppNotificationDevicesData.h"
+#include "Microsoft.Windows.AppNotifications.AppNotificationDevicesData.g.cpp"
+#include <IsWindowsVersion.h>
+
+namespace winrt::Microsoft::Windows::AppNotifications::implementation
+{
+    hstring AppNotificationDevicesData::CameraDeviceId()
+    {
+        return m_cameraDeviceId;
+    }
+
+    void AppNotificationDevicesData::CameraDeviceId(hstring const& value)
+    {
+        m_cameraDeviceId = value;
```

Comment:
You can check here if value is correct device id or not else throw error using devices enumeration API

Reply:
I would prefer to do a full-fledged validation on the system itself (shell/wnp) where the devices are connected, rather than performing a partial validation here. Even if the regex check passes, we cannot be certain that the device ID is valid.

Reply:
Yeah, we will be doing that check in shell for device id. I wanted to remove some garbage strings from WinAppSDK layer only like "12231231", "...............", these types of strings should now reach platform.
There is some format for device id, examples:
`\\?\USB#VID_045E&PID_0990&MI_00#6&db32c28&0&0000#{e5323777-f976-4f5b-9b55-b94699c46e44}\GLOBAL`
`\\?\SWD#MMDEVAPI#{0.0.1.00000000}.{a19be0b4-e6e9-404b-b3ae-e98dc182e850}#{2eef81be-33fa-4800-9670-1cd474972c3f}`

Reply:
As discussed, Validation is not required here. We will try to match this value with the list of enumerated devices on in the OS code, if it matches, we will display as selected in the drop-down else system default will be selected in the dropdown list.